dotCMS/src/main/java/com/dotcms/storage/GenerateMetadataConfig.java

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     * If true, means the medatada output will be stores in the memory cache.
+     */
+    private final boolean cache;
+
+    /**
+     * Cache key supplier, if cache is true
+     */
+    private final Supplier<String> cacheKeySupplier;
+
+    /**
+     * If true, means the metadata generated will be full except if there is any metaDataKeyFilter,
+     * only these fields will be accepted.
+     */
+    private final boolean full;
+
+    private GenerateMetadataConfig(final Builder builder) {
+
+        this.cache = builder.cache;
+        this.cacheKeySupplier = builder.cacheKeySupplier;
+        this.maxLength = builder.maxLength;
+        this.storageKey = builder.storageKey;
+        this.metaDataKeyFilter = builder.metaDataKeyFilter;
+        this.override = builder.override;
+        this.store = builder.store;
+        this.full = builder.full;
+    }
+
+    public StorageKey getStorageKey() {
+        return storageKey;
+    }
+
+    public boolean isStore() {
+        return store;
+    }
+
+    public boolean isOverride() {
+        return override;
+    }
+
+    public long getMaxLength() {
+        return maxLength;
+    }
+
+    public Predicate<String> getMetaDataKeyFilter() {
+        return metaDataKeyFilter;
+    }
+
+    public boolean isCache() {
+        return cache;
+    }
+
+    public Supplier<String> getCacheKeySupplier() {
+        return cacheKeySupplier;
+    }
+
+    public boolean isFull() {
+        return full;
+    }
+
+    public static final class Builder {
+
+        /**
+         * If true, means the metadata generated will be full except if there is any
+         * metaDataKeyFilter, only these fields will be accepted.
+         */
+        private boolean full;
+
+        /**
+         * Provides the supplier to stores the metadata generated, if store is true
+         */
+        private StorageKey storageKey;
+
+        /**
+         * If true, means the metadata output will be storage
+         */
+        private boolean store;
+
+        /**
+         * if store is true, and you want to force always the file generation set this to true (keep
+         * in mind it could be expensive)
+         */
+        private boolean override;
+
+        /**
+         * On huge file, we probably do not want to parse all the content, so this max will limited
+         * how much do we want to read
+         */
+        private long maxLength = FileStorageAPI.configuredMaxLength();
+
+        /**
+         * {@link Predicate} filter the meta data key for the map result generation
+         */
+        private Predicate<String> metaDataKeyFilter = s -> true; // no filter by default
+
+        /**
+         * If true, means the medatada output will be stores in the memory cache.
+         */
+        private boolean cache;
+
+        /**
+         * Cache key supplier, if cache is true
+         */
+        private Supplier<String> cacheKeySupplier = null;
```

Comment:
Codacy found an issue: [Avoid using redundant field initializer for 'cacheKeySupplier'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)